src/applications/vaos/components/layout/DetailPageLayout.jsx

```diff
@@ -202,31 +212,176 @@ function CancelButton({ appointment }) {
   return null;
 }
 
+export function TravelReimbursement({
+  appointmentDate,
+  claimData,
+  daysRemainingToFileClaim,
+  isPastAppointment,
+}) {
+  if (!isPastAppointment) return null;
+  if (!claimData) return null;
+
+  let body = null;
+
+  if (
+    claimData.message === TRAVEL_CLAIM_MESSAGES.noClaim &&
+    daysRemainingToFileClaim > 0
+  ) {
+    body = (
+      <>
+        <p className="vads-u-margin-y--0p5">
+          Days left to file: {daysRemainingToFileClaim}
+        </p>
+        <p className="vads-u-margin-y--0p5">
+          <va-link
+            data-testid="file-claim-link"
+            className="vads-u-margin-y--0p5"
+            href={`/appointments/claims/?date=${appointmentDate}`}
+            text="File a travel reimbursement claim"
+          />
+        </p>
+      </>
+    );
+  }
+  if (
+    claimData.message === TRAVEL_CLAIM_MESSAGES.noClaim &&
+    daysRemainingToFileClaim < 1
+  ) {
+    body = (
+      <>
+        <p className="vads-u-margin-y--0p5">
+          Days left to file: {daysRemainingToFileClaim}
+        </p>
+        <p className="vads-u-margin-y--0p5">
+          You didn’t file a claim for this appointment. You can only file for
+          reimbursement within 30 days of the appointment.
+        </p>
+        <p className="vads-u-margin-y--0p5">
+          <va-link
+            data-testid="how-to-file-claim-link"
+            className="vads-u-margin-y--0p5"
+            href="https://www.va.gov/resources/how-to-file-a-va-travel-reimbursement-claim-online/"
+            text="Learn more about travel reimbursement"
+          />
+        </p>
+      </>
+    );
+  }
+  if (claimData.id) {
+    body = (
+      <>
+        <p className="vads-u-margin-y--0p5">
+          You've already filed a claim for this facility and date.
+        </p>
+        <p className="vads-u-margin-y--0p5">
+          <va-link
+            data-testid="view-claim-link"
+            href={`/appointments/claims/${claimData.id}`}
+            text="Check your claim status"
+          />
+        </p>
+      </>
+    );
+  }
+
+  if (!body) return null;
+
+  return <Section heading="Travel reimbursement">{body}</Section>;
+}
+
+TravelReimbursement.propTypes = {
+  appointmentDate: PropTypes.string,
+  claimData: PropTypes.object,
+  daysRemainingToFileClaim: PropTypes.number,
+  isPastAppointment: PropTypes.bool,
+};
+
+export function AppointmentTasks({
+  appointmentDate,
+  claimData,
+  daysRemainingToFileClaim,
+  isPastAppointment,
+}) {
+  if (!isPastAppointment) return null;
```

Comment:
How much of this logic should be implemented in the frontend vs the backend? I imagine that the BE will also check if the appointment is in the past before adding `claimData` to the response so maybe this check is redundant and we only need to check for `claimData`/`claimData.message`?

Reply:
Also I believe this component only applies to "in person" appointments. How will that be determined and where (FE/BE) will that be implemented?

Reply:
@liztownd what do you think about this? I was assuming this logic would be on the front-end but maybe it makes more sense to do it all on the backend? I could then simplify things a bit by just checking if `travelPayClaim` exists.

Reply:
We have not been doing any checks on the backend code for if the appt is in the past. Because it's only being appended in the BE if the `include: claims` flag is `true` (like `avs`) the association logic should only be called if the flag is passed in, i.e. if it's a future appt the flag would be `false` so it shouldn't run on the BE for future appts. (Also, as to only applying to `in person` appts, we also do not do any checks for that on the BE, that would be handled on the FE.)

Reply:
@liztownd @JunTaoLuo In the list view fetch I know I'm fetching past appointments so I can append `claims` only when needed there. What would the response be for non-in-person appointments in that case? 

When a user hits refresh on an appointment details view though, I don't have a way of knowing what type of appointment I'm fetching so we are always appending `claims` to the api call in that case. I think after I get the appointment I'll have to check if it's past and in-person before I show anything.

Reply:
> Also I believe this component only applies to "in person" appointments. How will that be determined and where (FE/BE) will that be implemented?

I think I can move this to the `InPersonLayout` component and then check if it's a past appointment before displaying.

Reply:
Hmm a few things to consider:
1. Generally we're trying to avoid maintaining conditional logic on both the frontend and the backend, it just makes things harder to track. For AVS, the source of truth regarding past appointments is determined in the BE [here](https://github.com/department-of-veterans-affairs/vets-api/blob/f240922bc661aa2cb51af8907ae9a48fda18b79b/modules/vaos/app/services/vaos/v2/appointments_service.rb#L502). While I don't want to dictate the design since you're way more familiar with the implementation details, I would recommend the AVS pattern if that can work for your scenario.
2. I want to double check your "In-person" definition since we might be conflating different concepts. The "InPersonLayout" file you referenced is used for "clinic" appointments. However, there are other appointments such as Video at VA where the patient also needs to travel to a VA location to receive care. I want to check with you whether those should involve travel claims as well even though they will be rendered using the `VideoLayoutVA` file. If so we should actually take an inventory of all the appointment subtypes and see which ones claims are applicable to.

Reply:
Yes, I would defer to @JunTaoLuo as to utilizing the `avs` pattern - how you all decide to determine whether or not to call the appt > claim association logic should (IMO) be the same/similar. But, no, we won't do checking on appointments in that module, we just take the appointment(s) you send us and append matching claims (it's more involved than that, but I don't want to get too into the technical BE weeds on a FE PR 😁).

For item 2, yes, there are some Video visits that would qualify for travel pay, and some in person visits that _don't_. We're still working through all the different policies around it, but I love the idea of y'all taking an inventory of all the appointment types you might see then we can consult as to which ones would qualify and which ones wouldn't.